routes/application.js

```diff
@@ -0,0 +1,36 @@
+const application_db = require("../database/application_db")
+const student_db = require("../database/db")
+const program_db = require("../database/program_db")
+
+module.exports = {
+    getProgramListByStudentId: async function(studentId) {
+        let applicationList = await module.exports.activeApplications();
+        let program_list = []
+        for(let i = 0; i < applicationList.length; i++) {
+            if(applicationList[i].student == studentId) {
+                program_list.push(await program_db.getProgramById(applicationList[i].program))
+            }
+        }
+        return program_list
+    },
+    getStudentListByProgramId: async function(programId) {
+        let applicationList = await module.exports.activeApplications();
+        let student_list = []
+        for(let i = 0; i < applicationList.length; i++) {
+            if(applicationList[i].program == programId) {
+                student_list.push(await student_db.getStudentById(applicationList[i].student))
+            }
+        }
+        return student_list
+    },
+    activeApplications: async function() {
```

Comment:
instead of this, you can use [mongoose queries](https://mongoosejs.com/docs/queries.html). Actually, you can probably replace a decent chunk of this code with mongoose queries.

For example, to get all `Application` objects with a `status` that is `active`, you could add a function in the **application_db.js** file that does something like this:

```js
getActiveApplications: async function() {
    return await Application.find({'status': 'active'});
}
```